utils/aplay/aplay.c

```diff
@@ -326,20 +463,43 @@ static void *pcm_worker_routine(struct pcm_worker *w) {
 	size_t pause_counter = 0;
 	size_t pause_bytes = 0;
 
-	struct pollfd pfds[] = {{ w->ba_pcm_fd, POLLIN, 0 }};
+	int npfds = 0;
+	struct pollfd *pfds;
 	int timeout = -1;
 
+	if (mixer_device)
+		w->melem = init_mixer_control(&mixer, pcm_device, mixer_device);
```

Comment:
No, you cannot pass a PCM device id to init_mixer_control(). "default" is a special case, but try "hw:0,0" or "plughw:0,0" they give "Invalid CTL" error. You have to use a CTL device id, eg "hw:0" (assuming card 0 has a mixer), "bluealsa", etc.

In general, to achieve volume control with alsa you need 2 parameters, a control name (sometimes also referred to as a "mixer" name) and a control element name (eg "Master", etc). So to allow the user to select an arbitrary volume control, bluealsa-aplay would need 3 alsa params: PCM name, CTL name, and element name.

Reply:
Thanks for the review! I will update the code to use two parameters for the mixer control element. Or three with an additional index.